packages/frontend/core/src/blocksuite/ai/_common/chat-actions-handle.ts

```diff
@@ -273,17 +295,57 @@ const INSERT_BELOW = {
     const currentTextSelection = currentSelections.text;
     const currentBlockSelections = currentSelections.blocks;
     const currentImageSelections = currentSelections.images;
-    const [_, data] = host.command.exec(getSelectedBlocksCommand, {
-      currentTextSelection,
-      currentBlockSelections,
-      currentImageSelections,
-    });
-    if (!data.selectedBlocks) return false;
+
+    if (!host.doc.root) {
+      return false;
+    }
+
+    const [_, { isNothingSelected }] = host.command.exec(
+      isNothingSelectedCommand,
+      {
+        currentTextSelection,
+        currentBlockSelections,
+        currentImageSelections,
+      }
+    );
+
+    const currentMode = host.std.get(DocModeProvider).getEditorMode();
+
+    let selectedBlocks: BlockComponent[] = [];
+    if (isNothingSelected) {
+      let root = host.doc.root;
+      if (currentMode === 'edgeless') {
+        const gfx = host.std.get(GfxControllerIdentifier);
+        const selectedElements = gfx.selection.selectedElements;
+        const isOnlyOneNoteSelected =
+          selectedElements.length === 1 &&
+          selectedElements[0] instanceof NoteBlockModel;
+        if (isOnlyOneNoteSelected) {
+          root = selectedElements[0] as NoteBlockModel;
+        }
+      }
+      const [_, { lastBlock }] = host.command.exec(getLastContentBlockCommand, {
+        root,
+      });
+      selectedBlocks = lastBlock ? [lastBlock] : [];
```

Comment:
If the last block is a table block, will the content be inserted into a table cell?

Reply:
Insert below the last block means: the content will be inserted below the last content block. So, if the last is a table, content will be inserted below the table

Reply:
nsert below the last block means: the content will be inserted below the last content block. So, if the last is a table, content will be inserted below the table

Reply:
Insert below the last block means: the content will be inserted below the last content block. So, if the last is a table, content will be inserted below the table